src/lib/helper.js

```diff
@@ -3,16 +3,22 @@ const filter = (name) => {
   return name.replace(pattern, '').toLowerCase().trim();
 };
 
-const existingName = (doc, name) => {
-  const items = getItemsNamesFromDoc(doc);
-  let filteredName = filter(name);
-  return items.find((item) => filter(item) === filteredName);
+const existingName = (existingItems, newItemName) => {
+  const existingItemsNames = getItemsNamesFromDoc(existingItems);
+  let filteredNewItemName = filter(newItemName);
+  return existingItemsNames.find(
+    (item) => filter(item) === filteredNewItemName,
+  );
 };
 
-const getItemsNamesFromDoc = (doc) => {
-  let items = doc.data().items;
-  let itemsNames = items.map((item) => item.name);
-  return itemsNames;
+const getItemsNamesFromDoc = (items) => {
```

Comment:
Wow, great refactoring and improvement for getting the items' names!

Reply:
I will second Debora here! Awesome refactoring 👍